### ISSUE TITLE
move docker file from SpikeForest to SpikeSorter ?

### ISSUE THREAD

**Issue by samuelgarcia** [SpikeInterface/spikesorters#42]
As we discuss in Edimburgh docker file use in SF could be move to spikesorter repo.

Even this is totally useless for spiketoolkit tha main advantage could be that a sorter dev could make a unique PR in spikesorter when doing a new release of the sorter, (like Pierre, Cole and I did recently). At th moment, a sorter dev need to make 2 PR (one here and one in SF).

@magland : is it something simple or does it imply lot of works ?

**Comment by magland** [comment]
I agree that it is better to have the Dockerfile in the same location as the sorter. But it requires discussion on how to exactly do it.

**Comment by alejoe91** [comment]
@magland @samuelgarcia how should we do this exactly?

In my opinion, as the docker file is pretty small, we could add it in each spikesorter folder. Or we can have a docker folder in the github, so users would need to clone the repo or download the files. What do you guys think?

**Comment by magland** [comment]
I think we would need to have a planning discussion. There are various possible approaches, and we'd want to choose the best way to do it.

**Comment by magland** [comment]
I am currently working on this over here:
https://github.com/spikodrome/examples/tree/master/sorters

And I think I'd like to move that into spikeforest eventually. We can also echo something in spikeinterface.

The approach I am pursuing is to do spike sorting from the command line and to use an AutoRecordingExtractor inside the container. The AutoRecordingExtractor auto-detects the type of the input. See: https://github.com/spikodrome/spikodrome_utils/tree/master/spikodrome_utils/autoextractors

**Comment by samuelgarcia** [comment]
Hi Jeremy.
If I anderstand correctly your are doing a magic CLI for some sorters, right ?

I think this could be done directly inside spikesorters, no ?
In spikodrome, there is another copy theses dockers files.
In my opinion docker files should be centralize somewehre.
And if possible, next to sorter wrapper :  in spikesorter subfolder.

**Comment by magland** [comment]
I agree that everything should be in one place -- e.g., spikesorters. But there are various choices to be made, and I'm not at the stage where I have a clear idea of how it should go. For a first pass I am going to try to make CLI utilities that use AutoRecordingExtractor on the inside. The containerized sorters work on files, whereas the python sorters work on extractors.
